### PR TITLE
move footer to its own component

### PR DESCRIPTION
**Pull request by gonpombo8**
[Justin Sorry]. I have a toc with classes, makes it hard to follow. 

- [ ] refactor undoRedoBuffer to be a function instead of a class
- [ ] move footer to its own component

### CODE REVIEW

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const ONE_GB_IN_BYTES = ONE_MB_IN_BYTES * 1024
+
+export function UndoRedoArray(maxEntries: number, maxSize: number = ONE_GB_IN_BYTES) {
+  const valueList: UndoRedo[] = []
+  let memorySize: number = 0
+
+  function valueSize(val?: UndoRedo) {
+    let fileSize: number = 0
+    if (val?.$case === 'file' && val) {
+      for (const opreation of val.operations) {
```
Comment (nicoecheza) [inline packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:13]: `operation`

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      // Add current value size to the memory
+      const currentSize = valueSize(value)
+      memorySize += currentSize
+
+      // add the value to the list.
+      valueList.push(value)
+
+      // if we reach the max_size limit, start removing files.
+      while (memorySize >= maxSize) {
+        for (const index in valueList) {
```
Comment (nicoecheza) [inline packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:38]: `for in` doesn't ensure return order for arrays, I think we should use just a `for` loop (or `forEach`, etc)

Comment (nicoecheza) [reply packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:38]: and you will be able to avoid casting `index` to a number :P

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          const indexNumber: number = Number(index)
+          const size = valueSize(valueList[index])
+          if (size) {
+            memorySize -= size
+            valueList.splice(indexNumber, 1)
+            break
```
Comment (nicoecheza) [inline packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:44]: if you break here, you will just remove the first `$case === file` entry, and that could possible not be enough

**Review by nicoecheza — CHANGES_REQUESTED** [review]

**Comment by nicoecheza** [comment]
seems like some context menus are broken 😢 

Screen Shot 2023-05-25 at 10 33 25

Screen Shot 2023-05-25 at 10 33 31

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts (hunk 3 above)
Comment (gonpombo8) [reply packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:44]: nope, you have the while loop

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      memorySize -= valueSize(value)
+      return value
+    },
+    clear() {
+      memorySize = 0
+      valueList.length = 0
+    },
+    values() {
+      return Array.from(valueList)
+    },
```
[the remark this answers is not in the document]
Comment (gonpombo8) [reply packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:61]: if we do this, then you can mutate the valueList

File: packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts (hunk 2 above)
Comment (gonpombo8) [reply packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts:38]: mmm are u sure about the order? I think that was for some browsers in the past.
I'll change anyways but if this is the case then we have to refactor in a lot of places.

**Review by nicoecheza — APPROVED** [review]
